Clang 21: Move -Wno-character-conversion to WebKitCompilerFlags.cmake

rkirsling opened the pull request:
#### a8afa4e1b37ec9d0172382e40cd3c28106868d9c
<pre>
Clang 21: Move -Wno-character-conversion to WebKitCompilerFlags.cmake
<a href="https://bugs.webkit.org/show_bug.cgi?id=299720">https://bugs.webkit.org/show_bug.cgi?id=299720</a>

Reviewed by Michael Catanzaro.

I put this in OptionsPlayStation in 300663@main, but we don&apos;t need to pretend that it&apos;s platform-specific.

* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/WebKitCompilerFlags.cmake:

Canonical link: <a href="https://commits.webkit.org/300797@main">https://commits.webkit.org/300797@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/610cf61ca3cc302c7f9d44f07a22f74e40c5d16b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123914 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43629 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34326 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130701 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44353 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52223 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94246 "Failed to checkout and rebase branch from PR 51490") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126868 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35334 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/110846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74847 "Failed to checkout and rebase branch from PR 51490") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/34286 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74177 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/116073 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105062 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/29230 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133398 "Failed to checkout and rebase branch from PR 51490") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/122446 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50867 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38736 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/133398 "Failed to checkout and rebase branch from PR 51490") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51241 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/107066 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/133398 "Failed to checkout and rebase branch from PR 51490") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47886 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/26138 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47719 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19486 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50720 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/153542 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/50194 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/39023 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/53540 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/51868 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d2625cc0902b91f3e845d74c4c4e5c2d1d2d957c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51490

rkirsling commented:
Awesome, will plan to close this once that one gets r+. Thanks @mcatanzaro.

mcatanzaro commented on Source/cmake/WebKitCompilerFlags.cmake:
```diff
@@ -244,6 +244,11 @@ if (COMPILER_IS_GCC_OR_CLANG)
                                          -Werror=undefined-internal)
 endif ()
 
+# FIXME: https://bugs.webkit.org/show_bug.cgi?id=299689
+if (COMPILER_IS_CLANG AND ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 21)
+    WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-Wno-character-conversion)
+endif ()
```
```suggestion
# FIXME: https://bugs.webkit.org/show_bug.cgi?id=299689
WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-Wno-character-conversion)
```

I think we can safely do this unconditionally. This is a new warning, and we're simply not prepared for it yet. GCC doesn't implement it at all. If the compiler supports the warning, then we should turn it off until we're ready for it.

mcatanzaro replied:
```suggestion
# FIXME: https://bugs.webkit.org/show_bug.cgi?id=299689
WEBKIT_PREPEND_GLOBAL_COMPILER_FLAGS(-Wno-character-conversion)
```

I think we can safely do this unconditionally. This is a new warning, and we're simply not prepared for it yet. GCC doesn't implement it at all. If the compiler supports the warning, then we should turn it off until we're ready for it.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/610cf61ca3cc302c7f9d44f07a22f74e40c5d16b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51490

rkirsling committed:
Clang 21: Move -Wno-character-conversion to WebKitCompilerFlags.cmake
https://bugs.webkit.org/show_bug.cgi?id=299720

Reviewed by Michael Catanzaro.

I put this in OptionsPlayStation in 300663@main, but we don't need to pretend that it's platform-specific.

* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/WebKitCompilerFlags.cmake:

Canonical link: https://commits.webkit.org/300797@main
